packages/discord.js/src/managers/GuildManager.js

```diff
@@ -282,6 +283,71 @@ class GuildManager extends CachedManager {
     return data.reduce((coll, guild) => coll.set(guild.id, new OAuth2Guild(this.client, guild)), new Collection());
   }
 
+  /**
+   * @typedef {Object} FetchSoundboardSoundsOptions
+   * @param {Snowflake[]} guildIds The ids of the guilds to fetch soundboard sounds for
+   * @param {number} [time=10_000] The timeout for receipt of the soundboard sounds
+   */
+
+  /**
+   * Fetches soundboard sounds for the specified guilds.
+   * @param {FetchSoundboardSoundsOptions} options The options for fetching soundboard sounds
+   * @returns {Promise<Collection<Snowflake, Collection<Snowflake, SoundboardSound>>>}
+   * @example
+   * // Fetch soundboard sounds for multiple guilds
+   * const soundboardSounds = await client.guilds.fetchSoundboardSounds({
+   *  guildIds: ['123456789012345678', '987654321098765432'],
+   * })
+   *
+   * console.log(soundboardSounds.get('123456789012345678'));
+   */
+  async fetchSoundboardSounds({ guildIds, time = 10_000 }) {
+    const shardCount = await this.client.ws.getShardCount();
+    const shardIds = Map.groupBy(guildIds, guildId => ShardClientUtil.shardIdForGuildId(guildId, shardCount));
+
+    for (const [shardId, shardGuildIds] of shardIds) {
+      this.client.ws.send(shardId, {
+        op: GatewayOpcodes.RequestSoundboardSounds,
+        d: {
+          guild_ids: shardGuildIds,
+        },
+      });
+    }
```

Comment:
Call this after setting up the listener in the promise, otherwise you may miss events

Reply:
We do it the same way in `GuildMemberManager#_fetchMany()` and don’t have an issue, because all of this happens in the same event loop iteration, the ws responses will only be handled after